### PR TITLE
Refactor `quantum/command.{c,h}` for code size & {read,maintain}ability

### PR DESCRIPTION
**Pull request by liyang**
I accidentally a huge refactor. (This is part 1 of n.) It improves the code size & {read,maintain}ability significantly.

This PR reduces the firmware size by ~1.5kB on my ATmega32, when built with `CONSOLE_ENABLE` and `COMMAND_ENABLE` set to `yes`.

If possible, could this PR be merged rather than squashed?

## Description

Because the last commit moves `mousekey_console()` to `mousekey.c` (verbatim, trust me :), I recommend reading the penultimate commit separately to see the changes to `mousekey_console()` itself.

```
commit 3ec12a579518e1dda45b3163d042d4d781a6850c
Author: Liyang HU <git@liyang.hu>
Date:   Sun Feb 7 16:19:52 2021 +0000

    quantum/command.c: optimise `mousekey_console()` for size & legibility

    Made various tweaks to the interface, but still functionally identical.

    Assume `KC_1`…`KC_0` to be contiguous, and removed `numkey2num(…)` entirely.
    It was exported in `command.h` by 1a0bac8bcc for no obvious reason, before
    which it was `static`. I doubt anyone uses it.

    `mousekey_console()` is now enabled regardless of `MK_3_SPEED`.
    Needs fleshing out for things other than the X11 variant though.

    This commit saves 638 bytes on my ATmega32.

commit 003835288cb193ffda71671bb379bab716c01f2f
Author: Liyang HU <git@liyang.hu>
Date:   Mon Feb 8 11:58:48 2021 +0000

    quantum/command.c: replace `print(…); print_{,val_}{dec,hex}*(…);` sequences with single `xprintf(…)`

    `print_{dec,hex}*(…)` are just `#define`s for `xprintf(…)` anyway.

    Each additional `xprintf(…)` costs ~8 bytes: the call instructions,
    plus an additional NUL terminator.

    This _really_ adds up: this commit saves 814 bytes on my ATmega32.

commit ea79897bb1e957a69e5f26ce67c30996c7882051
Author: Liyang HU <git@liyang.hu>
Date:   Sun Feb 7 13:46:23 2021 +0000

    quantum/command.c: coalesce `print()`s in `command_common_help()` & `print_version()`

    Also undo some damage by clang-format in b624f32f94
```

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* ~~n/a~~

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] ~~My change requires a change to the documentation.~~
- [ ] ~~I have updated the documentation accordingly.~~
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] ~~I have added tests to cover my changes.~~
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by liyang** [comment]
Travis errored on something unrelated:
```
Making keebio/dsp40/rev1 with keymap default                                                           [ERRORS]
```

**Comment by spidey3** [comment]
This mostly looks good, but I would have preferred if the last change [a6576c5](https://github.com/qmk/qmk_firmware/pull/11842/commits/a6576c5ac7721c96bbcf50b4d5495dc04cad026e) were done in a separate PR.

**Comment by liyang** [comment]
> […] I would have preferred if the last change [a6576c5](https://github.com/qmk/qmk_firmware/pull/11842/commits/a6576c5ac7721c96bbcf50b4d5495dc04cad026e) were done in a separate PR.

Done!

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

**Review by Erovia — APPROVED** [review]
`__attribute__ ((weak))`

File: quantum/command.h [hunk 1]
```diff
@@ -28,8 +28,7 @@ bool command_extra(uint8_t code);
 bool command_console_extra(uint8_t code);
 
 #ifdef COMMAND_ENABLE
-uint8_t numkey2num(uint8_t code);
-bool    command_proc(uint8_t code);
+bool command_proc(uint8_t code);
```
Comment (fauxpark) [inline quantum/command.h:31]: I think clang-format will disagree with this (at least, the version of clang-format that the Docker image uses).

**Review by tzarc — APPROVED** [review]

**Pull request merged**